codefori/vscode-rpgle
# Initial implementation of IBM i symbol lookup functionality

Introduce a workspace symbol provider for IBM i, enabling symbol lookup based on the current editor and connection context. As well as a workspace symbol provider, we have also improved the implementation provider. You can right click on any word in the document, and use _go to implementation_ or _peek implementation_, which will initiate a lookup of that export in any binding directory in your library list. 
 
How to test workspace provider:

1. **Optional**: open a source member
2. Press Control+T/Command+T and wait for list of symbols to load. In theory, you should see a list of exports available to you based on your library list.
3. Select on an export and it should open the member to the line where it is defined.

How to test implementation provider: 

1. Open a source member which has references to prototypes defined with `extproc`
2. Right click on the procedure name and use _peek implementation(s)_ and expect to see the definition of the procedure if it is found.
